tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -135,7 +147,16 @@ public void CloseSpan(Span span)
 
             if (shouldPropagateMetadata)
             {
-                PropagateMetadata(spansToWrite);
+                if (spansToWrite.Array != null)
```

Comment:
I don't think we have to worry about the array being null
```suggestion
                if (spansToWrite.Count > 0)
```

Reply:
I was getting compiler warnings because `Array` is nullable 😛

Reply:
Yeah, this is the annoying thing with supporting older target frameworks. Because they only finished annotating the BCL in recent FX versions, we get erroneous null ref warnings.

I think judicious use of the dammit `!` operator in these cases is also fine

Reply:
oh, yeah. I forget `!` is a thing, thanks!